src/Kestrel.Core/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -26,15 +29,17 @@ private KestrelEventSource()
         // - Avoid renaming methods or parameters marked with EventAttribute. EventSource uses these to form the event object.
 
         [NonEvent]
-        public void ConnectionStart(IConnectionContext context, IConnectionInformation information)
+        public void ConnectionStart(ConnectionContext context)
         {
             // avoid allocating strings unless this event source is enabled
             if (IsEnabled())
             {
+                var info = context.Features.Get<IHttpConnectionFeature>();
```

Comment:
We can cache the IPEndPoint on FrameConnection and pass that directly to the Frame.